OpenRA.Mods.Common/Traits/Buildings/PrimaryBuilding.cs

```diff
@@ -82,5 +96,21 @@ public void SetPrimaryProducer(Actor self, bool state)
 
 			Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech", "PrimaryBuildingSelected", self.Owner.Faction.InternalName);
 		}
+
+		public IEnumerable<IRenderable> RenderAfterWorld(WorldRenderer wr)
+		{
+			if (!isPrimary)
+				yield break;
+
+			var tagImages = new Animation(wr.World, info.Image);
+			var pal = wr.Palette(info.Palette);
+			var tagxyOffset = new int2(0, 6);
```

Comment:
Why is this generated everytime new?
Can't we just have a const. or something like that?
(PS: What about unhardcoding this?)